### PR TITLE
chore: configure bundling to correct version of ES and browser support

### PR DESCRIPTION
**Pull request by mariuslundgard**
### Description

- Removes `./scripts/pkg-utils`
- Adds and configures `@sanity/pkg-utils` dependency
- Configures package.json, tsconfig.json and tsconfig.lib.json
- Removes all unused config files (e.g. .npmignore)
- Builds packages targeting `es2017` and browsers used by > 0.2% of people

### What to review

- That the builds produce the expected results
- Test consuming `sanity` in a 3rd party repository (we need to release a canary version)
- Check that the `SANITY_VERSION` export has the correct value:

```sh
# install the canary version
npm i sanity@3.0.0-v3-pkg-utils.59
```

```ts
import {SANITY_VERSION} from 'sanity'
console.log(SANITY_VERSION)
```

### CODE REVIEW

**Review by stipsan — APPROVED** [review]
LGTM 👍

File: packages/sanity/package.json [hunk 1]
```diff
@@ -78,18 +84,27 @@
       ]
     }
   },
+  "files": [
+    "_internal.js",
+    "_unstable.js",
+    "cli.js",
+    "desk.js",
+    "form.js",
```
[the remark this answers is not in the document]
Comment (mariuslundgard) [reply packages/sanity/package.json:92]: These are generated when running `yarn build`. They're ignored by `.gitignore`, but should be included in the npm tarball (webpack v3 still use these, but we can remove them at some point).

File: packages/sanity/package.json [hunk 2]
```diff
@@ -13,51 +13,57 @@
...
     "content"
   ],
+  "bin": {
+    "sanity": "./bin/sanity"
+  },
   "source": "./src/_exports/index.ts",
-  "main": "./lib/cjs/_exports/index.js",
-  "module": "./lib/esm/_exports/index.js",
+  "main": "./lib/index.cjs",
+  "module": "./lib/index.js",
```
[the remark this answers is not in the document]
Comment (mariuslundgard) [reply packages/sanity/package.json:21]: Hm, I had some issues using `"type": "module"` – will give it one last shot :)

File: test/config.js [hunk 3]
```diff
@@ -62,11 +64,12 @@ exports.createJestConfig = function createJestConfig({
...
-        'babel-jest',
-        // rootMode upwards makes use of the global babel.config.js
-        {rootMode: 'upward'},
-      ],
+      '^.+\\.tsx?$': ['esbuild-jest', {sourcemap: true}],
+      // '\\.[jt]sx?$': [
+      //   'babel-jest',
+      //   // rootMode upwards makes use of the global babel.config.js
+      //   {rootMode: 'upward'},
+      // ],
```
[the remark this answers is not in the document]
Comment (mariuslundgard) [reply test/config.js:72]: Yes!

[the remark this answers is not in the document]
Comment (mariuslundgard) [reply test/config.js:72]: There's some issues with the tests still. Will address.

[the remark this answers is not in the document]
Comment (mariuslundgard) [reply test/config.js]: Fixed by reverting back to `babel-jest` 👍

File: packages/sanity/package.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (mariuslundgard) [reply packages/sanity/package.json:21]: Hm, I'm scared of including that here. Let's make a separate story for that.

**Pull request merged**